cmd/offlinesync/offlinesync.go

```diff
@@ -16,8 +17,30 @@ import (
 	"github.com/spf13/viper"
 )
 
-// Run executes the sync-offline-activity command.
-func Run(v *viper.Viper) (int, error) {
+// RunWithoutRateLimiting executes the sync-offline-activity command without rate limiting.
+func RunWithoutRateLimiting(v *viper.Viper) (int, error) {
+	paramOffline := params.LoadOfflineParams(v)
+
+	return Run(v, paramOffline)
+}
+
+// RunWithRateLimiting executes sync-offline-activity command with rate limiting enabled.
+func RunWithRateLimiting(v *viper.Viper) (int, error) {
+	paramOffline := params.LoadOfflineParams(v)
+
+	if cmdheartbeat.RateLimited(paramOffline) {
+		return exitcode.Success, nil
```

Comment:
Maybe adding a debug message to say it's rate limited and heartbeats weren't synced at this time?